pkg/cable/metrics.go

```diff
@@ -59,28 +63,115 @@ var (
 			remoteHostnameLabel,
 		},
 	)
+	connectionsGauge = prometheus.NewGaugeVec(
+		prometheus.GaugeOpts{
+			Name: "connections",
+			Help: "Number of connections and corresponding status (by cable driver and cable)",
+		},
+		[]string{
+			cableDriverLabel,
+			localClusterLabel,
+			localHostnameLabel,
+			localEndpointIpLabel,
+			remoteClusterLabel,
+			remoteHostnameLabel,
+			remoteEndpointIpLabel,
+			connectionsStatusLabel,
+		},
+	)
+	connectionEstablishedTimestampGauge = prometheus.NewGaugeVec(
+		prometheus.GaugeOpts{
+			Name: "connection_established_timestamp",
+			Help: "Timestamp of last successful connection established (by cable driver and cable)",
+		},
+		[]string{
+			cableDriverLabel,
+			localClusterLabel,
+			localHostnameLabel,
+			localEndpointIpLabel,
+			remoteClusterLabel,
+			remoteHostnameLabel,
+			remoteEndpointIpLabel,
+		},
+	)
+	connectionLatencySecondsGauge = prometheus.NewGaugeVec(
+		prometheus.GaugeOpts{
+			Name: "connection_latency_seconds",
+			Help: "Connection latency in seconds (average RTT, by cable driver and cable)",
+		},
+		[]string{
+			cableDriverLabel,
+			localClusterLabel,
+			localHostnameLabel,
+			localEndpointIpLabel,
+			remoteClusterLabel,
+			remoteHostnameLabel,
+			remoteEndpointIpLabel,
+		},
+	)
 )
 
 func init() {
-	prometheus.MustRegister(rxGauge)
+	prometheus.MustRegister(rxGauge, txGauge, connectionsGauge, connectionEstablishedTimestampGauge, connectionLatencySecondsGauge)
 }
 
-func RecordRxBytes(cableDriverName string, localEndpoint, remoteEndpoint *submv1.EndpointSpec, bytes int) {
-	rxGauge.With(prometheus.Labels{
+func getLabels(cableDriverName string, localEndpoint, remoteEndpoint *submv1.EndpointSpec) prometheus.Labels {
+	return prometheus.Labels{
+		cableDriverLabel:      cableDriverName,
+		localClusterLabel:     localEndpoint.ClusterID,
+		localHostnameLabel:    localEndpoint.Hostname,
+		localEndpointIpLabel:  localEndpoint.PublicIP,
+		remoteClusterLabel:    remoteEndpoint.ClusterID,
+		remoteHostnameLabel:   remoteEndpoint.Hostname,
+		remoteEndpointIpLabel: remoteEndpoint.PublicIP,
+	}
+}
+
+func getTxRxLabels(cableDriverName string, localEndpoint, remoteEndpoint *submv1.EndpointSpec) prometheus.Labels {
+	return prometheus.Labels{
 		cableDriverLabel:    cableDriverName,
```

Comment:
Regarding your point about whether the tx/rx gauges should have the same labels as the other metrics, you’re right, they should; so we should use `getLabels()` everywhere, and drop `getTxRxLabels()`.